src/AzureClient/Extensions.cs

```diff
@@ -22,5 +30,96 @@ public static void AddAzureClient(this IServiceCollection services)
         {
             services.AddSingleton<IAzureClient, AzureClient>();
         }
+
+        /// <summary>
+        ///      Encapsulates a given <c>AzureClientError</c> as the result of an execution.
+        /// </summary>
+        /// <param name="azureClientError">
+        ///      The result of an IAzureClient API call.
+        /// </param>
+        public static ExecutionResult ToExecutionResult(this AzureClientError azureClientError) =>
+            new ExecutionResult
+            {
+                Status = ExecuteStatus.Error,
+                Output = azureClientError.ToDescription()
+            };
+
+        /// <summary>
+        ///     Returns the string value of the <c>DescriptionAttribute</c> for the given
+        ///     <c>AzureClientError</c> enumeration value.
+        /// </summary>
+        /// <param name="azureClientError"></param>
+        /// <returns></returns>
+        public static string ToDescription(this AzureClientError azureClientError)
+        {
+            var attributes = azureClientError
+                .GetType()
+                .GetField(azureClientError.ToString())
+                .GetCustomAttributes(typeof(DescriptionAttribute), false) as DescriptionAttribute[];
+            return attributes?.Length > 0 ? attributes[0].Description : string.Empty;
+        }
+
+        /// <summary>
+        ///      Encapsulates a given <c>AzureClientError</c> as the result of an execution.
+        /// </summary>
+        /// <param name="task">
+        ///      A task which will return the result of an IAzureClient API call.
+        /// </param>
+        public static async Task<ExecutionResult> ToExecutionResult(this Task<AzureClientError> task) =>
+            (await task).ToExecutionResult();
+
+        internal static Table<JobDetails> ToJupyterTable(this JobDetails jobDetails) =>
+            new List<JobDetails> { jobDetails }.ToJupyterTable();
+
+        internal static Table<JobDetails> ToJupyterTable(this IEnumerable<JobDetails> jobsList) =>
+            new Table<JobDetails>
```

Comment:
This may be difficult for a Python client to pick up and deserialize, since it looks like a list of the form `[{"JobId": jobId}, {"JobName", jobName}, ...]`. Given how frequently you're using object-as-a-table patterns here, it may be worth making a custom result encoder that adapts the JSON serialization output to your usecase.

Reply:
Yes, this is a very good point. This will need to change.

---
In reply to: [427472915](https://github.com/microsoft/iqsharp/pull/144#discussion_r427472915) [](ancestors = 427472915)

Reply:
I've opened a task to track this work.